README.md
# leaflet-compare

A Leaflet control to add a split screen to compare two map overlays.

**This project is a fork of the fork ([leaflet-splitmap](https://github.com/QuantStack/leaflet-splitmap)) of the [leaflet-side-by-side](https://github.com/digidem/leaflet-side-by-side) plugin**

![screencast example](screencast.gif)

### L.control.compare(_leftLayer[s]_, _rightLayer[s]_, options)

Creates a new Leaflet Control for comparing two layers or collections of layers. It does
not add the layers to the map - you need to do that manually. Extends `L.Control` but
`setPosition()` and `getPosition()` are used differently compared to a normal `L.Control`
object. Currently `setPosition(offset)` sets the position of the slider by a given
offset between 0 and 1. `getPosition()` returns the position of the slider in pixels.
The latter can be converted to a slider offset between 0 and 1 by dividing by
`map.getSize().x`. 

### Parameters

| parameter     | type           | description   |
| ----------    | -------------- | ------------- |
| `leftLayers`  | L.Layer\|array | A Leaflet Layer or array of layers to show on the left side of the map. Any layer added to the map that is in this array will be shown on the left |
| `rightLayers` | L.Layer\|array | A Leaflet Layer or array of layers to show on the right side of the map. Any layer added to the map that is in this array will be shown on the right. These *should not be* the same as any layers in `leftLayers` |
| `options`     | Object         | Options |
| `options.thumbSize` | Number     | Width of the slider thumb in pixels. Defaults to `44` |
| `options.padding` | Number     | Padding between slider min/max and the edge of the screen in pixels. Defaults to `0` |
| `options.position` | Number    | Initial position of the slider given by a number between `0` (left/min) and `1` (right/max) for. Defaults to `0.5` |

### Events

Subscribe to events using [these methods](http://leafletjs.com/reference.html#events)

| Event         | Data           | Description   |
| ----------    | -------------- | ------------- |
| `dividermove` | {x: Number} | Fired when the divider is moved. Returns an event object with the property `x` = the pixels of the divider from the left side of the map container. |


### Example

[Live Example](https://phloose.github.io/leaflet-compare/) see [source](index.html)

### Installation

`leaflet-compare` is compatible with Leaflet 1.7.1 but should also work with older
versions.

To install it you can use npm:

`npm install leaflet-compare`

Then import it in your project:

`import "leaflet-compare"`

This will assign the `Compare` class to the `L.Control` namespace and the factory
function `compare` to the `L.control` namespace. 

As an alternative you can import the `Compare` class directly:

`import { Compare } from "leaflet-compare"`

Be sure to have included the Leaflet css and js files and the plugin's files
`leaflet-compare.css` *before* you use the plugin. Otherwise the slider will not be
shown.

#### Usage in the browser without ES6 imports

You can use `leaflet-compare` from a CDN like [unpkg](https://unpkg.com/). Include the
plugin's css and js files *after* the links to the Leaflet files. Then either use
`myScript.js` or use script tags in the body of the referencing html page.

```html
<head>
    <link rel="stylesheet" href="https://unpkg.com/leaflet@1.7.1/dist/leaflet.css" />
    <script src="https://unpkg.com/leaflet@1.7.1/dist/leaflet.js" ></script>
    <link rel="stylesheet" href="https://unpkg.com/leaflet-compare/dist/leaflet-compare.css" />
    <script src="https://unpkg.com/leaflet-compare/dist/leaflet-compare.js"></script>
    <script src="myScript.js"></script>
</head>
```

### Limitations

- The divider is not movable with IE.
- Probably won't work in IE8, but what does?

### License

MIT


CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file.
This project adheres to [Semantic Versioning](http://semver.org/).


## [leaflet-compare 1.0.1] - 2021-02-01

### Changed

- Republish to overcome npm's unpublish policy. Version 1.0.0 had the fork's links
  in package.json, some files where not ignored in .npmignore and the release date was
  still tba.

## [leaflet-compare 1.0.0] - 2021-02-01

### Added

- New option `position` for setting the sliders initial position.
- `setPosition` is no longer a noop and serves the purpose of setting the range sliders position.

### Changed

- Use webpack-dev-server instead of budo
- Use ESLint instead of standard
- Use jest for testing
- `leaflet` is now a peerDependency since it is assumed that the user already has leaflet installed himself.

### Fixed

- Use `L.Evented` instead of deprecated `L.Mixin.Events`

### Security

- Updated all devDependencies to their latest versions
- Removed vulnerable (and unused) dependencies from the original leaflet-side-by-side
  plugin.

## [leaflet-splitmap]

- ADDED: Allows Leaflet version 0.7.7 through 1.x

## [leaflet-side-by-side v2.0.0] - 2015-12-08

- ADDED: Add `setLeftLayers()` and `setRightLayers()` methods
- ADDED: `options.padding`
- ADDED: `getPosition()` returns the x coordinate (relative to the map container) of the divider
- FIXED: **[BREAKING]** Export factory function on `L.control` not `L.Control`
- FIXED: Slider drag was not working on touch devices

## [leaflet-side-by-side v1.1.1] - 2015-12-03

- FIXED: fix package.json settings for npm distribution

## [leaflet-side-by-side v1.1.0] - 2015-12-03

- ADDED: Events
- FIXED: Fix initial divider position in Firefox, should start in middle of map

## [leaflet-side-by-side] v1.0.2 - 2015-12-02

Initial release


__tests__/leaflet-compare.spec.js
import L from "leaflet";
import { Compare } from "../src/leaflet-compare";

function thumbAndPaddingOffset(position, thumbSize = 42, padding = 0) {
    return (0.5 - position) * (2 * padding + thumbSize);
}

describe("leaflet-compare", () => {
    let map;
    let leftlayer;
    let rightlayer;
    let compare;

    const mapHeight = 600;
    const mapWidth = 800;

    beforeEach(() => {
        const mapDiv = document.createElement("div");
        // We use a virtual DOM so the clientHeight/clientWidth will be zero. Define
        // them so the map gets a fake viewport.
        Object.defineProperty(mapDiv, "clientHeight", {
            configurable: true,
            value: mapHeight,
        });
        Object.defineProperty(mapDiv, "clientWidth", {
            configurable: true,
            value: mapWidth,
        });
        map = L.map(mapDiv, {
            renderer: new L.SVG(), // from: https://github.com/oliverroick/Leaflet.Deflate/blob/master/tests/L.Deflate.test.js
            center: [0, 0],
            zoom: 10,
        });
    });

    afterEach(() => {
        compare.remove();
        map.remove();
    });
    describe("Methods", () => {
        describe("#setPosition", () => {
            test("via constructor", () => {
                compare = new Compare(leftlayer, rightlayer, {
                    position: 0.7,
                }).addTo(map);
                expect(compare.getPosition() / map.getSize().x).toBeCloseTo(
                    0.7,
                    1,
                );
            });

            test("via method", () => {
                compare = new Compare(leftlayer, rightlayer)
                    .addTo(map)
                    .setPosition(0.7);
                expect(compare.getPosition() / map.getSize().x).toBeCloseTo(
                    0.7,
                    1,
                );
            });
        });
    });

    describe("Layers with pane='tilePane' (GridLayer, TileLayer)", () => {
        beforeEach(() => {
            leftlayer = L.gridLayer().addTo(map);
            rightlayer = L.gridLayer().addTo(map);
        });
        test("default slider position", () => {
            compare = new Compare(leftlayer, rightlayer).addTo(map);

            expect(leftlayer.getContainer().style.clip).toEqual(
                `rect(0px, ${mapWidth / 2}px, ${mapHeight}px, 0px)`,
            );
            expect(rightlayer.getContainer().style.clip).toEqual(
                `rect(0px, ${mapWidth}px, ${mapHeight}px, ${mapWidth / 2}px)`,
            );
        });
        test("slider position on the left (position < 0.5)", () => {
            const position = 0.3;
            compare = new Compare(leftlayer, rightlayer)
                .addTo(map)
                .setPosition(position);
            expect(leftlayer.getContainer().style.clip).toEqual(
                `rect(0px, ${
                    mapWidth * position + thumbAndPaddingOffset(position)
                }px, ${mapHeight}px, 0px)`,
            );
            expect(rightlayer.getContainer().style.clip).toEqual(
                `rect(0px, ${mapWidth}px, ${mapHeight}px, ${
                    mapWidth * position + thumbAndPaddingOffset(position)
                }px)`,
            );
        });
        test("slider position on the right (position > 0.5)", () => {
            const position = 0.7;
            compare = new Compare(leftlayer, rightlayer)
                .addTo(map)
                .setPosition(position);
            expect(leftlayer.getContainer().style.clip).toEqual(
                `rect(0px, ${
                    mapWidth * 0.7 + thumbAndPaddingOffset(0.7)
                }px, ${mapHeight}px, 0px)`,
            );
            expect(rightlayer.getContainer().style.clip).toEqual(
                `rect(0px, ${mapWidth}px, ${mapHeight}px, ${
                    mapWidth * 0.7 + thumbAndPaddingOffset(0.7)
                }px)`,
            );
        });
    });
    describe("Layers with pane='overlayPane' (Point, Polygon, Polyline, ImageOverlay, LayerGroup, FeatureGroup, GeoJSON)", () => {
        beforeEach(() => {
            map.createPane("left");
            map.createPane("right");

            leftlayer = L.circle([0, -20], {
                pane: "left",
                radius: 1000,
            }).addTo(map);
            rightlayer = L.circle([0, 20], {
                pane: "right",
                radius: 1000,
            }).addTo(map);
        });
        test("default slider position", () => {
            compare = new Compare(leftlayer, rightlayer).addTo(map);
            expect(leftlayer.getPane().style.clip).toEqual(
                `rect(0px, ${mapWidth / 2}px, ${mapHeight}px, 0px)`,
            );
            expect(rightlayer.getPane().style.clip).toEqual(
                `rect(0px, ${mapWidth}px, ${mapHeight}px, ${mapWidth / 2}px)`,
            );
        });
        test("slider position on the left (position < 0.5)", () => {
            const position = 0.2;
            compare = new Compare(leftlayer, rightlayer)
                .addTo(map)
                .setPosition(position);
            expect(leftlayer.getPane().style.clip).toEqual(
                `rect(0px, ${
                    mapWidth * position + thumbAndPaddingOffset(position)
                }px, ${mapHeight}px, 0px)`,
            );
            expect(rightlayer.getPane().style.clip).toEqual(
                `rect(0px, ${mapWidth}px, ${mapHeight}px, ${
                    mapWidth * position + thumbAndPaddingOffset(position)
                }px)`,
            );
        });
        test("slider position on the right (position > 0.5)", () => {
            const position = 0.8;
            compare = new Compare(leftlayer, rightlayer)
                .addTo(map)
                .setPosition(position);
            expect(leftlayer.getPane().style.clip).toEqual(
                `rect(0px, ${
                    mapWidth * position + thumbAndPaddingOffset(position)
                }px, ${mapHeight}px, 0px)`,
            );
            expect(rightlayer.getPane().style.clip).toEqual(
                `rect(0px, ${mapWidth}px, ${mapHeight}px, ${
                    mapWidth * position + thumbAndPaddingOffset(position)
                }px)`,
            );
        });
    });
});


.eslintrc.js
module.exports = {
    env: {
        browser: true,
        es6: true,
    },
    extends: ["airbnb-base", "eslint:recommended", "prettier"],
    plugins: ["prettier"],
    parserOptions: {
        ecmaVersion: 2018,
        sourceType: "module",
    },
    overrides: [
        {
            files: ["**/*.spec.js"],
            env: {
                jest: true,
            },
        },
    ],
    rules: {
        "prettier/prettier": [
            "error",
            {
                endOfLine: "auto",
                tabWidth: 4,
                arrowParens: "avoid",
                trailingComma: "all",
            },
        ],
        quotes: ["error", "double"],
        indent: [
            "error",
            4,
            {
                SwitchCase: 1,
            },
        ],
        "no-console": "off",
        "linebreak-style": "off",
        "no-unused-expressions": [
            "error",
            {
                allowShortCircuit: true,
            },
        ],
        "arrow-parens": ["error", "as-needed"],
        "func-names": "off",
        "no-plusplus": [
            "error",
            {
                allowForLoopAfterthoughts: true,
            },
        ],
        "no-param-reassign": [
            "error",
            {
                props: false,
            },
        ],
        "no-underscore-dangle": [
            "error",
            {
                allowAfterThis: true,
            },
        ],
    },
};


webpack.config.js
const path = require("path");
const MiniCssExtractPlugin = require("mini-css-extract-plugin");
const { version } = require("./package.json");

module.exports = {
    // embeddable jupyter-leaflet bundle
    entry: "./src/index.js",
    output: {
        filename: "leaflet-compare.js",
        path: path.resolve(__dirname, "dist"),
        publicPath: "/dist/",
    },
    devtool: "source-map",
    module: {
        rules: [
            {
                test: /\.css$/,
                use: [MiniCssExtractPlugin.loader, "css-loader"],
            },
            {
                test: /\.(jpg|png|gif|svg)$/,
                use: [
                    {
                        loader: "url-loader",
                        options: {
                            limit: 8000,
                            name: "images/[hash]-[name].[ext]",
                        },
                    },
                ],
            },
        ],
    },
    externals: {
        leaflet: "L",
    },
    devServer: {
        index: "./index.html",
        publicPath: "./src",
        writeToDisk: true,
    },
    plugins: [
        new MiniCssExtractPlugin({
            filename: "leaflet-compare.css",
        }),
    ],
};


src/leaflet-compare.js
import L from "leaflet";

let mapWasDragEnabled;
let mapWasTapEnabled;

// Leaflet v0.7 backwards compatibility
function on(el, types, fn, context) {
    types.split(" ").forEach(type => {
        L.DomEvent.on(el, type, fn, context);
    });
}

// Leaflet v0.7 backwards compatibility
function off(el, types, fn, context) {
    types.split(" ").forEach(type => {
        L.DomEvent.off(el, type, fn, context);
    });
}

function getRangeEvent(rangeInput) {
    return "oninput" in rangeInput ? "input" : "change";
}

function cancelMapDrag() {
    mapWasDragEnabled = this._map.dragging.enabled();
    mapWasTapEnabled = this._map.tap && this._map.tap.enabled();
    this._map.dragging.disable();
    this._map.tap && this._map.tap.disable();
}

function uncancelMapDrag(e) {
    this._refocusOnMap(e);
    if (mapWasDragEnabled) {
        this._map.dragging.enable();
    }
    if (mapWasTapEnabled) {
        this._map.tap.enable();
    }
}

// convert arg to an array - returns empty array if arg is undefined
function asArray(arg) {
    if (typeof arg === "undefined") {
        return [];
    }
    if (Array.isArray(arg)) {
        return arg;
    }
    return [arg];
}

L.Control.Compare = L.Control.extend({
    options: {
        thumbSize: 42,
        padding: 0,
        position: 0.5,
    },

    initialize(leftLayers, rightLayers, options) {
        this._leftLayers = asArray(leftLayers);
        this._rightLayers = asArray(rightLayers);
        this._updateClip();
        L.setOptions(this, options);
    },

    getPosition() {
        const rangeValue = this._range.value;
        const offset =
            (0.5 - rangeValue) *
            (2 * this.options.padding + this.options.thumbSize);
        return this._map.getSize().x * rangeValue + offset;
    },

    setPosition(offset) {
        if (!this._map) {
            return this;
        }
        this._range.value = offset;
        this._updateClip();
        return this;
    },

    includes: L.Evented.prototype,

    addTo(map) {
        this.remove();
        this._map = map;
        this._container = L.DomUtil.create(
            "div",
            "leaflet-sbs",
            // eslint-disable-next-line no-underscore-dangle
            map._controlContainer,
        );
        this._divider = L.DomUtil.create(
            "div",
            "leaflet-sbs-divider",
            this._container,
        );
        this._range = L.DomUtil.create(
            "input",
            "leaflet-sbs-range",
            this._container,
        );
        this._range.type = "range";
        this._range.min = 0;
        this._range.max = 1;
        this._range.step = "any";
        this._range.value = 0.5;
        // eslint-disable-next-line no-multi-assign
        this._range.style.paddingLeft = this._range.style.paddingRight = `${this.options.padding}px`;
        this._addEvents();
        this._updateClip();
        if (this.options.position) {
            this.setPosition(this.options.position);
        }
        return this;
    },

    remove() {
        if (!this._map) {
            return this;
        }
        this._leftLayers.forEach(leftLayer => {
            if (leftLayer.getContainer) {
                leftLayer.getContainer().style.clip = "";
            } else {
                leftLayer.getPane().style.clip = "";
            }
        });

        this._rightLayers.forEach(rightLayer => {
            if (rightLayer.getContainer) {
                rightLayer.getContainer().style.clip = "";
            } else {
                rightLayer.getPane().style.clip = "";
            }
        });
        this._removeEvents();
        L.DomUtil.remove(this._container);
        this._map = null;
        return this;
    },

    _updateClip() {
        if (!this._map) {
            return this;
        }
        const map = this._map;
        const nw = map.containerPointToLayerPoint([0, 0]);
        const se = map.containerPointToLayerPoint(map.getSize());
        const clipX = nw.x + this.getPosition();
        const dividerX = this.getPosition();
        this._divider.style.left = `${dividerX}px`;
        this.fire("dividermove", { x: dividerX });
        const clipLeft = `rect(${[nw.y, clipX, se.y, nw.x].join("px,")}px)`;
        const clipRight = `rect(${[nw.y, se.x, se.y, clipX].join("px,")}px)`;

        this._leftLayers.forEach(leftLayer => {
            if (leftLayer.getContainer) {
                leftLayer.getContainer().style.clip = clipLeft;
            } else {
                leftLayer.getPane().style.clip = clipLeft;
            }
        });

        this._rightLayers.forEach(rightLayer => {
            if (rightLayer.getContainer) {
                rightLayer.getContainer().style.clip = clipRight;
            } else {
                rightLayer.getPane().style.clip = clipRight;
            }
        });
        return this;
    },

    _addEvents() {
        const range = this._range;
        const map = this._map;
        if (!map || !range) return;
        map.on("move", this._updateClip, this);
        map.on("layeradd layerremove", this._updateLayers, this);
        on(range, getRangeEvent(range), this._updateClip, this);
        on(
            range,
            L.Browser.touch ? "touchstart" : "mousedown",
            cancelMapDrag,
            this,
        );
        on(
            range,
            L.Browser.touch ? "touchend" : "mouseup",
            uncancelMapDrag,
            this,
        );
    },

    _removeEvents() {
        const range = this._range;
        const map = this._map;
        if (range) {
            off(range, getRangeEvent(range), this._updateClip, this);
            off(
                range,
                L.Browser.touch ? "touchstart" : "mousedown",
                cancelMapDrag,
                this,
            );
            off(
                range,
                L.Browser.touch ? "touchend" : "mouseup",
                uncancelMapDrag,
                this,
            );
        }
        if (map) {
            map.off("layeradd layerremove", this._updateLayers, this);
            map.off("move", this._updateClip, this);
        }
    },
});

L.control.compare = function (leftLayers, rightLayers, options) {
    return new L.Control.Compare(leftLayers, rightLayers, options);
};

export const { Compare } = L.Control;
export default Compare;


src/leaflet-splitmap.js
import L from "leaflet";

let mapWasDragEnabled;
let mapWasTapEnabled;

// Leaflet v0.7 backwards compatibility
function on(el, types, fn, context) {
    types.split(" ").forEach(type => {
        L.DomEvent.on(el, type, fn, context);
    });
}

// Leaflet v0.7 backwards compatibility
function off(el, types, fn, context) {
    types.split(" ").forEach(type => {
        L.DomEvent.off(el, type, fn, context);
    });
}

function getRangeEvent(rangeInput) {
    return "oninput" in rangeInput ? "input" : "change";
}

function cancelMapDrag() {
    mapWasDragEnabled = this._map.dragging.enabled();
    mapWasTapEnabled = this._map.tap && this._map.tap.enabled();
    this._map.dragging.disable();
    this._map.tap && this._map.tap.disable();
}

function uncancelMapDrag(e) {
    this._refocusOnMap(e);
    if (mapWasDragEnabled) {
        this._map.dragging.enable();
    }
    if (mapWasTapEnabled) {
        this._map.tap.enable();
    }
}

// convert arg to an array - returns empty array if arg is undefined
function asArray(arg) {
    if (typeof arg === "undefined") {
        return [];
    }
    if (Array.isArray(arg)) {
        return arg;
    }
    return [arg];
}

L.Control.SplitMap = L.Control.extend({
    options: {
        thumbSize: 42,
        padding: 0,
        position: null,
    },

    initialize(leftLayers, rightLayers, options) {
        this._leftLayers = asArray(leftLayers);
        this._rightLayers = asArray(rightLayers);
        this._updateClip();
        L.setOptions(this, options);
    },

    getPosition() {
        const rangeValue = this._range.value;
        const offset =
            (0.5 - rangeValue) *
            (2 * this.options.padding + this.options.thumbSize);
        return this._map.getSize().x * rangeValue + offset;
    },

    setPosition(offset) {
        if (!this._map) {
            return this;
        }
        this._range.value = offset;
        this._updateClip();
        return this;
    },

    includes: L.Evented.prototype,

    addTo(map) {
        this.remove();
        this._map = map;
        this._container = L.DomUtil.create(
            "div",
            "leaflet-sbs",
            // eslint-disable-next-line no-underscore-dangle
            map._controlContainer,
        );
        this._divider = L.DomUtil.create(
            "div",
            "leaflet-sbs-divider",
            this._container,
        );
        this._range = L.DomUtil.create(
            "input",
            "leaflet-sbs-range",
            this._container,
        );
        this._range.type = "range";
        this._range.min = 0;
        this._range.max = 1;
        this._range.step = "any";
        this._range.value = 0.5;
        // eslint-disable-next-line no-multi-assign
        this._range.style.paddingLeft = this._range.style.paddingRight = `${this.options.padding}px`;
        this._addEvents();
        this._updateClip();
        if (this.options.position) {
            this.setPosition(this.options.position);
        }
        return this;
    },

    remove() {
        if (!this._map) {
            return this;
        }
        this._leftLayers.forEach(leftLayer => {
            if (leftLayer.getContainer) {
                leftLayer.getContainer().style.clip = "";
            } else {
                leftLayer.getPane().style.clip = "";
            }
        });

        this._rightLayers.forEach(rightLayer => {
            if (rightLayer.getContainer) {
                rightLayer.getContainer().style.clip = "";
            } else {
                rightLayer.getPane().style.clip = "";
            }
        });
        this._removeEvents();
        L.DomUtil.remove(this._container);
        this._map = null;
        return this;
    },

    _updateClip() {
        if (!this._map) {
            return this;
        }
        const map = this._map;
        const nw = map.containerPointToLayerPoint([0, 0]);
        const se = map.containerPointToLayerPoint(map.getSize());
        const clipX = nw.x + this.getPosition();
        const dividerX = this.getPosition();
        this._divider.style.left = `${dividerX}px`;
        this.fire("dividermove", { x: dividerX });
        const clipLeft = `rect(${[nw.y, clipX, se.y, nw.x].join("px,")}px)`;
        const clipRight = `rect(${[nw.y, se.x, se.y, clipX].join("px,")}px)`;

        this._leftLayers.forEach(leftLayer => {
            if (leftLayer.getContainer) {
                leftLayer.getContainer().style.clip = clipLeft;
            } else {
                leftLayer.getPane().style.clip = clipLeft;
            }
        });

        this._rightLayers.forEach(rightLayer => {
            if (rightLayer.getContainer) {
                rightLayer.getContainer().style.clip = clipRight;
            } else {
                rightLayer.getPane().style.clip = clipRight;
            }
        });
        return this;
    },

    _addEvents() {
        const range = this._range;
        const map = this._map;
        if (!map || !range) return;
        map.on("move", this._updateClip, this);
        map.on("layeradd layerremove", this._updateLayers, this);
        on(range, getRangeEvent(range), this._updateClip, this);
        on(
            range,
            L.Browser.touch ? "touchstart" : "mousedown",
            cancelMapDrag,
            this,
        );
        on(
            range,
            L.Browser.touch ? "touchend" : "mouseup",
            uncancelMapDrag,
            this,
        );
    },

    _removeEvents() {
        const range = this._range;
        const map = this._map;
        if (range) {
            off(range, getRangeEvent(range), this._updateClip, this);
            off(
                range,
                L.Browser.touch ? "touchstart" : "mousedown",
                cancelMapDrag,
                this,
            );
            off(
                range,
                L.Browser.touch ? "touchend" : "mouseup",
                uncancelMapDrag,
                this,
            );
        }
        if (map) {
            map.off("layeradd layerremove", this._updateLayers, this);
            map.off("move", this._updateClip, this);
        }
    },
});

L.control.splitMap = function (leftLayers, rightLayers, options) {
    return new L.Control.SplitMap(leftLayers, rightLayers, options);
};

export const { SplitMap } = L.Control;
export default SplitMap;
